models/h265_codec_configuration.go

```diff
@@ -0,0 +1,71 @@
+	QP                       *int64                     `json:"qp,omitempty"`
+	MaxBitrate               *int64                     `json:"maxBitrate,omitempty"`
+	MinBitrate               *int64                     `json:"minBitrate,omitempty"`
+	BufSize                  *int64                     `json:"bufsize,omitempty"`
+	MinGOP                   *int64                     `json:"minGop,omitempty"`
+	MaxGOP                   *int64                     `json:"maxGop,omitempty"`
+	Level                    bitmovintypes.H265Level    `json:"level,omitempty"`
+	RCLookahead              *int64                     `json:"rcLookahead,omitempty"`
+	BAdapt                   bitmovintypes.BAdapt       `json:"bAdapt,omitempty"`
+	MaxCTUSize               bitmovintypes.MaxCTUSize   `json:"maxCTUSize,omitempty"`
+	TUIntraDepth             bitmovintypes.TUIntraDepth `json:"tuIntraDepth,omitempty"`
+	TUInterDepth             bitmovintypes.TUInterDepth `json:"tuInterDepth,omitempty"`
+	MotionSearch             bitmovintypes.MotionSearch `json:"motionSearch,omitempty"`
+	SubMe                    *int64                     `json:"subMe,omitempty"`
+	MotionSearchRange        *int64                     `json:"motionSearchRange,omitempty"`
+	WeightPredictionOnPSlice *bool                      `json:"weightPredictionOnPSlice,omitempty"`
+	WeightPredictionOnBSlice *bool                      `json:"weightPredictionOnBSlice,omitempty"`
+	SAO                      *bool                      `json:"sao,omitempty"`
+}
+
+type H265CodecConfigurationData struct {
+	//Success fields
+	Result   H265CodecConfiguration `json:"result,omitempty"`
+	Messages []Message              `json:"messages,omitempty"`
+
+	//Error fields
+	Code             *int64   `json:"code,omitempty"`
+	Message          *string  `json:"message,omitempty"`
+	DeveloperMessage *string  `json:"developerMessage,omitempty"`
+	Links            []Link   `json:"links,omitempty"`
+	Details          []Detail `json:"details,omitempty"`
+}
+
+type H265CodecConfigurationResponse struct {
+	RequestID *string                      `json:"requestId,omitempty"`
+	Status    bitmovintypes.ResponseStatus `json:"status,omitempty"`
+	Data      H265CodecConfigurationData   `json:"data,omitempty"`
+}
+
+type H265CodecConfigurationListResult struct {
+	TotalCount *int64                   `json:"totalCount,omitempty"`
+	Previous   *string                  `json:"previous,omitempty"`
+	Next       *string                  `json:"next,omitempty"`
+	Items      []H265CodecConfiguration `json:"items,omitempty"`
+}
+
+type H265CodecConfigurationListData struct {
+	Result H265CodecConfigurationListResult `json:"result,omitempty"`
+}
+
+type H265CodecConfigurationListResponse struct {
+	RequestID *string                        `json:"requestId,omitempty"`
+	Status    bitmovintypes.ResponseStatus   `json:"status,omitempty"`
+	Data      H265CodecConfigurationListData `json:"data,omitempty"`
```

Comment:
I think at some point we might want to get rid of all these and just go with a `interface{}` in the response

Reply:
Same for the ListResult.. We could simplify all of these to be one struct.. But I'll do that refactoring at a later point :)